File: _tools/_garbage/scratch/explore_magic.py
import os
import json

base_dir = r"e:\AntigravitiProgetti\CompendioTheWitcher\_tools\src-packs\MAGIA_E_MALEDIZIONI"

results = []

for root, dirs, files in os.walk(base_dir):
    for file in files:
        if file.endswith('.json'):
            file_path = os.path.join(root, file)
            try:
                with open(file_path, 'r', encoding='utf-8') as f:
                    data = json.load(f)
                name = data.get("name", "Unknown")
                item_type = data.get("type", "Unknown")
                rel_path = os.path.relpath(file_path, base_dir)
                results.append({
                    "name": name,
                    "type": item_type,
                    "rel_path": rel_path,
                    "file": file
                })
            except Exception as e:
                print(f"Error reading {file_path}: {e}")

print(f"Found {len(results)} assets.")
# Print some stats and first few results
for r in results[:10]:
    print(r)

# Save results to a json file to read easily
with open(r"e:\AntigravitiProgetti\CompendioTheWitcher\scratch\magic_assets.json", "w", encoding="utf-8") as f:
    json.dump(results, f, indent=4, ensure_ascii=False)

File: _tools/_garbage/scratch/explore_magic.js
const fs = require('fs');
const path = require('path');

const baseDir = "e:\\AntigravitiProgetti\\CompendioTheWitcher\\_tools\\src-packs\\MAGIA_E_MALEDIZIONI";
const results = [];

function walk(dir) {
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        const fullPath = path.join(dir, file);
        const stat = fs.statSync(fullPath);
        if (stat && stat.isDirectory()) {
            walk(fullPath);
        } else if (file.endsWith('.json')) {
            try {
                const data = JSON.parse(fs.readFileSync(fullPath, 'utf8'));
                const name = data.name || "Unknown";
                const type = data.type || "Unknown";
                const relPath = path.relative(baseDir, fullPath);
                results.push({
                    name: name,
                    type: type,
                    relPath: relPath,
                    file: file
                });
            } catch (e) {
                console.error("Error reading file:", fullPath, e);
            }
        }
    });
}

walk(baseDir);
console.log(`Found ${results.length} assets.`);
fs.writeFileSync("e:\\AntigravitiProgetti\\CompendioTheWitcher\\scratch\\magic_assets.json", JSON.stringify(results, null, 4), 'utf8');
console.log("Results saved to magic_assets.json");
